core/assay.js

```diff
@@ -0,0 +1,129 @@
+import harden from '@agoric/harden';
+
+import { insist } from '../util/insist';
+import { mustBeSameStructure, mustBeComparable } from '../util/sameStructure';
+
+// This module treats labels as black boxes. It is not aware
+// of issuers, and so can handle labels whose issuers are merely
+// presences of remote issuers.
+
+// `makeAssayMaker` takes in a 'strategy' object which defines set
+// operations for a particular kind of assay. For instance, for the
+// natAssay, these operations are arithmetic.
+
+// Return an assay, which makes amounts, validates amounts, and
+// provides set operations over amounts. An amount is a pass-by-copy
+// description of some set of erights. An amount has a label and a
+// quantity. All amounts made by the same assay have the same label
+// but differ in quantity.
+//
+// An assay is pass-by-presence, but is not designed to be usefully
+// passed. Rather, we expect each vat that needs to operate on amounts
+// will have its own local assay to do so.
+
+function makeAssay(label, strategy) {
+  mustBeComparable(label);
+
+  // The brand represents recognition of the amount as authorized.
+  const brand = new WeakSet();
+
+  const assay = harden({
+    getLabel() {
+      return label;
+    },
+
+    getStrategy() {
```

Comment:
Now that this is mostly settled, would you update the .chainmail file? 

I think my attitude is that the .chainmail file should focus on the customer API rather than internal details. `getStrategy()` is visible to the API, so it should at least appear in that file. 

OTOH, strategies are a crucial part of the extension story, so the interface they present and while is relied on by other parts of the code needs to be written up somewhere. That could either be in one of the .js files (assays? issuers? I'm not sure) or in the .chainmail file if it makes more sense to segregate all the documentation there.

Reply:
Sure, that sounds good. I made an issue for this